SD-K9/src/app/components/outdoor-map/outdoor-map.component.ts

```diff
@@ -81,69 +92,132 @@ export class OutdoorMapComponent implements OnInit, OnDestroy, Map {
       }
 
       this.map = new google.maps.Map(this.mapElement.nativeElement, mapOptions);
+
+      //update or load from google map state
       if (this.data.id && !this._routeStore.getRoute(this.data.id)) {
         this._googleStore.storeMap({id: this.data.id, google: google, map: this.map, route: false});        // new map state
-      } 
+      }
       else {
         this._googleStore.updateGoogleMap({id: this.data.id, google: google, map: this.map, route: true}); // reload old map state
-      } 
-
-      var markerIcon = {
-        url: 'assets/images/map-pin.png',
-        scaledSize: new google.maps.Size(25, 44),
-        origin: new google.maps.Point(0,0),
-        anchor: new google.maps.Point(12, 44)
-      };
+      }
 
+      //add a marker on the current position
       this.userMarker = new google.maps.Marker({
         position: this.currentPos,
         map: this.map,
         title: 'You are here',
-        icon: markerIcon
+        icon: {
+          url: 'assets/images/map-pin.png',
+          scaledSize: new google.maps.Size(25, 44),
+          origin: new google.maps.Point(0,0),
+          anchor: new google.maps.Point(12, 44)
+        }
       });
 
-      this.drawBuildings();
+      //draw buildings overlay from config file
+      this._drawBuildings();
+
+      //check if we should show/hide the building markers when the user zooms in/out
+      let _self = this;
+      google.maps.event.addListener(this.map, 'zoom_changed', function() {
+          _self._hideShowMarkers(_self);
+      });
```

Comment:
I'm not too sure I understand this, would it be easier to just always have the buildings marked, regardless of whether or not the map is zoomed in or out?

Reply:
I think the idea is that if you are zoomed out and the building are tiny, then the labels shouldn't show. I may be wrong though.

Reply:
Yes, the labels don't resize so if you zoom out a lot and don't hide them they overflow from the buildings. This way the labels always stay inside the building outlines